src/Core/Managed/Shared/TelemetryClientExtensions.cs

```diff
@@ -57,9 +61,22 @@ public static IOperationHolder<T> StartOperation<T>(this TelemetryClient telemet
                 operationTelemetry.Context.Operation.ParentId = parentOperationId;
             }
 
+            if (correlationContext != null)
+            {
+                foreach (var item in correlationContext)
```

Comment:
We're merging dictionaries here. Perhaps that could be extracted into a helper method. Consider an extension method on ```IDictionary<string, string>``` so you can write:
```C#
operationTelemetry.Context.CorrelationContext.Merge(correlationContext);